aws/resource_aws_acm_certificate.go

```diff
@@ -227,7 +225,7 @@ func resourceAwsAcmCertificateCreateImported(d *schema.ResourceData, meta interf
 func resourceAwsAcmCertificateCreateRequested(d *schema.ResourceData, meta interface{}) error {
 	acmconn := meta.(*AWSClient).acmconn
 	params := &acm.RequestCertificateInput{
-		DomainName:       aws.String(strings.TrimSuffix(d.Get("domain_name").(string), ".")),
+		DomainName:       aws.String(trimTrailingPeriod(d.Get("domain_name").(string))),
```

Comment:
Similar note as above, we should prefer to do nothing here instead of custom logic.

```suggestion
		DomainName:       aws.String(d.Get("domain_name").(string)),
```